crates/orderbook/src/api/order_validation.rs

```diff
@@ -336,11 +362,35 @@ impl OrderValidating for OrderValidator {
             )
             .await
             .unwrap_or(false)
+        {
+            return Ok((order, unsubsidized_fee));
+        }
+
+        if !self
+            .balance_fetcher
+            .has_sufficient_balance(order_creation.sell_token, owner, min_balance)
+            .await
+            .unwrap_or(false)
         {
             return Err(ValidationError::InsufficientFunds);
```

Comment:
uber-naming-nit: This should probably be renamed to `InsufficientBalance` now.

Reply:
Thanks @nlordell for notifying, indeed is relevant. 

https://github.com/gnosis/cowswap/pull/2010